services/users.js

```diff
@@ -21,7 +21,6 @@ async function signup (firstName, lastName, email) {
 }
 
 async function login (secret) {
-  // TODO: throw error when secret is unrecognized
   return UserModel.getUserBySecret(secret)
 }
```

Comment:
You don't get to delete the TODO if you're not throwing an error anywhere. You need to check the return value of `UserModel.getUserBySecret` and throw an error if it's empty. Otherwise, the route never catches an error, and thus never returns 404. 

Did you test this manually?

Reply:
From what I see in the console, it throws an error when its empty and the api catches it. It used to return a 500 status but I changed it to a 404 . Is this ok or do you want a manual check in the `UserModel.getUserBySecret` and a distinction between a 404 and 500?

Reply:
Manual check in the service level please. Distinction between 404 would be great

Reply:
What should the 500 status do?

Reply:
Should it also route to the 404 page? If left alone, it will just display the static addfriend page

Reply:
Yeah, for now it should probably redirect to a 500 page. I think we'll want to move towards a more robust solution later -- ideally we'd render the 404/500 without changing the URL -- but that requires [fancier state management](https://www.newline.co/@3nvi/centralizing-api-error-handling-in-react-apps--80296494) to avoid code duplication.

Reply:
Okok sounds good